Adding support for CGC w/ Automation

jordan-rash opened the pull request:
CC: @adthrasher

adthrasher commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,59 @@
+        uses: docker/login-action@v1
+        with:
+          registry: cgc-images.sbgenomics.com
+          username: ${{ secrets.CGC_USERNAME }}
+          password: ${{ secrets.CGC_TOKEN }}
```
CGC_TOKEN and CGC_USERNAME has been added

adthrasher commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,59 @@
+      - 
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - 
+        name: Login to GitHub Container Registry
+        uses: docker/login-action@v1 
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.CLOUDY_USERNAME }}
+          password: ${{ secrets.CLOUDY_TOKEN }}
```
```suggestion
```
Since NetBID has a separate Docker image, we will just build and deploy to CGC.

adthrasher replied:
```suggestion
          username: ${{ secrets.GH_USERNAME }}
          password: ${{ secrets.GH_TOKEN }}
```

adthrasher commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,59 @@
+          file: ./Dockerfile.cgc
+          push: true
+          tags: |
+            ghcr.io/${{ env.REPO_LOWER }}:latest
+            ghcr.io/${{ env.REPO_LOWER }}:${{ env.SHORT_SHA }}
+            ghcr.io/${{ env.REPO_LOWER }}:${{ github.event.release.tag_name }}
```
```suggestion
```

adthrasher commented on .github/workflows/publish.yml:
```diff
@@ -0,0 +1,59 @@
+            ghcr.io/${{ env.REPO_LOWER }}:latest
+            ghcr.io/${{ env.REPO_LOWER }}:${{ env.SHORT_SHA }}
+            ghcr.io/${{ env.REPO_LOWER }}:${{ github.event.release.tag_name }}
+            cgc-images.sbgenomics.com/${{ env.REPO_LOWER }}:latest
+            cgc-images.sbgenomics.com/${{ env.REPO_LOWER }}:${{ env.SHORT_SHA }}
+            cgc-images.sbgenomics.com/${{ env.REPO_LOWER }}:${{ github.event.release.tag_name }}
```
```suggestion
            cgc-images.sbgenomics.com/stjude/netbid:latest
            cgc-images.sbgenomics.com/stjude/netbid:${{ env.SHORT_SHA }}
            cgc-images.sbgenomics.com/stjude/netbid:${{ github.event.release.tag_name }}
```

adthrasher commented on cgc/netbid.cwl:
```diff
@@ -0,0 +1,100 @@
+                "glob": "$(inputs.expression_set.nameroot).log"
+            }
+        }
+    ],
```
```suggestion
    ],
    "doc": "# Description\n\nNetBID is a data-driven system biology pipeline using a data-driven network-based Bayesian inference approach to find drivers from transcriptomics, proteomics, or phosphoproteomics data. The drivers can be either transcription factors (TF) or signaling factors (SIG).\n\nNetBID2 has the following key steps to perform hidden driver analysis:\n1.\tActivity calculation of drivers based on drivers’ regulons from a pre-built or user-provided SJARACNe network;\n2.\tDiscovery of differential expressed genes and differential activated drivers;\n3.\tGeneration of the master table for drivers;\n4.\tVisualizing drivers with significance profiles and target genes.\n\n# Inputs and outputs of NetBID workflow\n## Inputs:\n*\tExpression matrix - comma-delimited expression matrix file with columns as samples, rows as genes.\n*\tMetadata file - comma-delimited file with two columns: sample and group.\n*\tSignaling (SIG) network - file with each row an edge from the SIG network constructed using SJARACNe (https://github.com/jyyulab/SJARACNe)\n*\tTranscription factor (TF) network - file with each row an advantage from the TF network constructed using SJARACNe (https://github.com/jyyulab/SJARACNe)\n\n## Outputs:\n*\tExcel file with differential expressed genes and differential activated drivers\n*\tPicture file visualizing drivers with significance profiles\n\n# Common issues\n*\tThe first row and the first column of the expression matrix file must be sample names and gene names, respectively.\n*\tThe metadata file must have at least two sample groups in the 2nd column.",

```

adthrasher commented on cgc/netbid.cwl:
```diff
@@ -0,0 +1,100 @@
+                "shellQuote": false,
+                "position": 1
+            },
+            "label": "Gene Expression Matrix"
```
```suggestion
            "label": "Gene Expression Matrix",
            "doc": "comma-delimited expression matrix file with columns as samples, rows as genes."

```

adthrasher commented on cgc/netbid.cwl:
```diff
@@ -0,0 +1,100 @@
+                "shellQuote": false,
+                "position": 1
+            },
+            "label": "Transcription Factor Network"
```
```suggestion
            "label": "Transcription Factor Network",
            "doc": "file with each row an edge from the TF network constructed using SJARACNe (https://github.com/jyyulab/SJARACNe)"

```

adthrasher commented on cgc/netbid.cwl:
```diff
@@ -0,0 +1,100 @@
+                "shellQuote": false,
+                "position": 1
+            },
+            "label": "Signaling Network"
```
```suggestion
            "label": "Signaling Network",
            "doc": "file with each row an edge from the SIG network constructed using SJARACNe (https://github.com/jyyulab/SJARACNe)"

```

adthrasher commented on cgc/netbid.cwl:
```diff
@@ -0,0 +1,100 @@
+                "shellQuote": false,
+                "position": 3
+            },
+            "label": "Sample Grouping File"
```
```suggestion
            "label": "Sample Grouping File",
            "doc": "comma-delimited file with two columns: sample and group."
```

adthrasher commented on cgc/netbid.cwl:
```diff
@@ -0,0 +1,100 @@
+    "cwlVersion": "v1.2",
+    "baseCommand": [],
+    "inputs": [
+        {
+            "id": "rscript",
+            "type": "File",
+            "inputBinding": {
+                "shellQuote": false,
+                "position": 0
+            }
+        },
```
```suggestion
```

adthrasher commented on cgc/netbid.cwl:
```diff
@@ -0,0 +1,100 @@
+        },
+        {
+            "class": "DockerRequirement",
+            "dockerPull": "cgc-images.sbgenomics.com/cjrash/stjude/netbid:latest"
```
```suggestion
            "dockerPull": "cgc-images.sbgenomics.com/stjude/netbid:latest"
```

adthrasher commented on .github/workflows/publish.yml:
```diff
@@ -23,13 +23,6 @@ jobs:
-        with:
-          registry: ghcr.io
-          username: ${{ secrets.CLOUDY_USERNAME }}
-          password: ${{ secrets.CLOUDY_TOKEN }}
```
From yesterday's discussion, we are going to need to revert this change so that we can push the image to ghcr. We should update the secret names to `GH_USERNAME` and `GH_TOKEN`. I've asked Adam to create those secrets on the repo.

adthrasher replied:
@jordan-rash

adthrasher commented on .github/workflows/publish.yml:
```diff
@@ -39,9 +39,6 @@ jobs:
           tags: |
-            ghcr.io/${{ env.REPO_LOWER }}:latest
-            ghcr.io/${{ env.REPO_LOWER }}:${{ env.SHORT_SHA }}
-            ghcr.io/${{ env.REPO_LOWER }}:${{ github.event.release.tag_name }}
```
We'll need to revert this change also after yesterday's discussion.

adthrasher replied:
@jordan-rash

jordan-rash commented:
@adthrasher I just rebased and squashed the commits while dropping the GH change.....I am reviewing now, but another set of eyes is helpful....when I use the "Commit Changes" button in the UI, the commit log is not helpful, so I might have missed a piece

adthrasher reviewed (approved):
Just the small change to the environment secrets, then it looks good.

jordan-rash commented on .github/workflows/cgc.yml:
```diff
@@ -0,0 +1,29 @@
+    name: Deploy
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set the docker tag from Repo Tag
+        id: set_dockertag
+        env:
+          IMAGE: cgc-images.sbgenomics.com/stjude/netbid
+          VERSION_TAG: ${{ github.event.release.tag_name }}
+        run: |
+          jq --arg image "${{ env.IMAGE }}:${{ env.VERSION_TAG }}" '(.requirements | .[] | select(.class == ("DockerRequirement")) | .dockerPull) |= $image' cgc/netbid.cwl > cgc/netbid.cwl.new
+          mv cgc/netbid.cwl.new cgc/netbid.cwl
+          cat cgc/netbid.cwl
```
@adthrasher Ran this locally via `act` and it looks to work okay.  let me know what you think

adthrasher replied:
I think that looks good.

adamdingliang reviewed (approved):
Completed sanity check
